theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -312,254 +309,29 @@ def infer_shape(self, node, xshp):
 
     def c_code(self, node, name, inputs, outputs, sub):
         return """
+        PyGpuArrayObject *%(name)s_tmp;
+        %(name)s_tmp = PyArray_GETCONTIGUOUS(%(inp)s);
```

Comment:
What do you mean by copy data as-is? It receive the stride. So it don't
need access to the ndarray object.

On Thu, Oct 1, 2015 at 4:20 PM, abergeron <notifications@github.com> wrote:

> In theano/sandbox/gpuarray/basic_ops.py
> <https://github.com/Theano/Theano/pull/3476#discussion_r40961234>:
>
> > @@ -312,254 +309,29 @@ def infer_shape(self, node, xshp):
> >
> >      def c_code(self, node, name, inputs, outputs, sub):
> >          return """
> > +        PyGpuArrayObject *%(name)s_tmp;
> > +        %(name)s_tmp = PyArray_GETCONTIGUOUS(%(inp)s);
>
> Because pygpu_fromhostdata() does not have access to the ndarray object
> and currently tries to copy the data as-is. I've realized that it's not a
> good idea and made a ticket to change that in Theano/libgpuarray#90
> <https://github.com/Theano/libgpuarray/issues/90>
>
> I made a ticket that might help at
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3476/files#r40961234>.
>